spec.html

```diff
@@ -50,21 +49,18 @@ <h1>GetWaiterList ( _block_, _i_ )</h1>
     </emu-alg>
   </emu-clause>
 
-  <emu-clause id="sec-alarm-functions">
-    <h1>Alarm Functions</h1>
-    <p><ins>This is a new section.</ins></p>
-    <p>An alarm function is an anonymous built-in function that has [[WaiterList]], [[Waiter]], [[Kind]], and [[Result]] internal slots.</p>
-    <p>When an alarm function is called with no arguments, the following steps are taken:</p>
+  <emu-clause id="sec-triggertimeout" aoid="TriggerTimeout">
+    <h1>TriggerTimeout( _WL_, _waiterRecord_ )</h1>
+    <p><ins>This is a new abstract operation.</ins></p>
+    <p>The abstract operation TriggerTimeout takes two arguments, a WaiterList _WL_ and a Record _waiterRecord_. It performs the following steps:</p>
     <emu-alg>
-      1. Let _F_ be the active function object.
-      1. Assert: _F_ has a [[WaiterList]] internal slot whose value is a WaiterList.
-      1. Assert: _F_ has a [[WaiterRecord]] internal slot whose value is a Record.
-      1. Let _WL_ be _F_.[[WaiterList]].
-      1. Let _waiterRecord_ be _F_.[[WaiterRecord]].
-      1. Set _waiterRecord_.[[Result]] to `"timed-out"`.
+      1. Assert: _waiterRecord_.[[Timeout]] is finite.
       1. Perform EnterCriticalSection(_WL_).
-      1. Perform RemoveWaiter(_WL_, _waiterRecord_).
-      1. Perform NotifyWaiter(_WL_, _waiterRecord_).
+      1. If _waiterRecord_ is on the list of waiters in _WL_, then
+          1. Set _waiterRecord_.[[Result]] to `"timed-out"`.
+          1. Perform RemoveWaiter(_WL_, _waiterRecord_).
+          1. Perform NotifyWaiter(_WL_, _waiterRecord_).
+          1. NOTE: No alarm that subsequently triggers for _waiterRecord_ (in the concurrent thread referenced in AddWaiter) will have any effect. If alarms are implemented with thunks, the thunk associated with this alarm is now dead and can be reclaimed; any scheduled timeout associated with the alarm can be cancelled.
```

Comment:
What's an "alarm" at this point? Maybe you could phrase this as, "NOTE: No invocation of this abstract algorithm (from the parallel block in AddWaiter) will have any effect, and any associated resources can be reclaimed." (But I'm not sure if we need this note at all.)

Reply:
Hm, sure, I can delete this note.